src/backends/pytorch/libtorch_backend.cc

```diff
@@ -613,12 +669,36 @@ LibTorchBackend::Context::Run(
   // Run...
   RETURN_IF_ERROR(Execute(&inputs_, &outputs_));
 
-  // Make sure each output is of the expected size and copy it into
-  // the payload responses.
+  // TODO Validate Output shape after model execution
+
   for (const auto& output : base->Config().output()) {
     const std::string& name = output.name();
     std::string index_str = name.substr(name.find(deliminator) + 2);
+    int op_index = std::atoi(index_str.c_str());
+    int max_index = outputs_.size() - 1;
+    if ((op_index < 0) || (op_index > max_index)) {
+      return Status(
+          RequestStatusCode::INVALID_ARG,
+          "The output " + name +
+              " in the model config refers to an output index which doesn't "
```

Comment:
Haven't we already tested (in the provider_utils) that the request asks only for outputs that are in the model config. That means that here we just have to make sure that every output required by the request is actually provided by the result. Isn't it ok if the result provides extra outputs that the request doesn't need (and/or that are listed in the model config). Perhaps unexpected? Perhaps we can log an error? But the request can still complete.

Reply:
Yes we test that the outputs asked for are present in the model config. However we do not test that the outputs have a valid *__\<index\> i.e. an the model config can have *_1 but the model can have only one output i.e. *__0. 
However now I see that these checks should be present in the earlier validation section as they need only be performed once. I will move it there.